tests/test_cache.py

```diff
@@ -0,0 +1,61 @@
+import numba
+import numpy as np
+import pytest
+
+from stumpy import cache, stump
+
+if numba.config.DISABLE_JIT:
+    pytest.skip("Skipping Tests JIT is disabled", allow_module_level=True)
+
+
+def test_cache_save():
+    def get_cache_fnames_ref():
+        cache._clear()
+        cache._enable()
+        stump(np.random.rand(10), 3)
+        cache_data_fnames = [
+            fname for fname in cache._get_cache() if fname.endswith(".nbc")
+        ]
+        cache_index_fnames = [
+            fname for fname in cache._get_cache() if fname.endswith(".nbi")
+        ]
+        cache._clear()
+        return cache_data_fnames, cache_index_fnames
+
+    def get_cache_fnames_comp():
+        cache._clear()
+        cache._save()
+        stump(np.random.rand(10), 3)
+        cache_data_fnames = [
+            fname for fname in cache._get_cache() if fname.endswith(".nbc")
+        ]
+        cache_index_fnames = [
+            fname for fname in cache._get_cache() if fname.endswith(".nbi")
+        ]
+        cache._clear()
+        return cache_data_fnames, cache_index_fnames
+
+    ref_data, ref_index = get_cache_fnames_ref()
```

Comment:
What is the code difference between `get_cache_fnames_ref()` and `get_cache_fnames_comp()`?

Reply:
One is using `cache._enable()` and the other is using `cache.save`. I am going to revise it to improve readability.

Reply:
I see. I'm still failing to see the point of what we are trying to test? `cache._save()` calls `cache._enable()` anyways so it should be expected that they produce the same result. That is, unless you believe that `cache._recompile()` might change the saving?

Reply:
This was to make sure if users can just replace `cache._enable()` with `cache.save()` once they notice the warning in `cache._enable()`.

> so it should be expected that they produce the same result. That is, unless you believe that cache._recompile() might change the saving?

Their only difference is in the `nbi` files. In face, in the first call of `cache.save()`, the `nbi` files of all njit functions will be saved into the numba cache directory.

Reply:
@seanlaw 
I revised the test function to make it more readable.. but I was thinking more about what you said:

>  I'm still failing to see the point of what we are trying to test? cache._save() calls cache._enable() anyways 

I think I am testing a trivial thing here... I mean.... the whole process is the same in ref and comp, and I am just replacing `cache._enable` with `cache.save`. So, I am now inclined to remove it in the next commits.

Reply:
Sounds good